tests/uv/test_uv.py

```
@@ -360,3 +360,69 @@ def test_check_call_no_hint_for_pip_install_uv(capsys):
+
+
+def test_parse_req_file_vcs_with_inline_comment_strips_only_comment(tmp_path):
+    # The trickiest case: a VCS spec with a fragment AND a trailing comment.
+    # Comment is preceded by whitespace so it must be stripped; fragment is
+    # part of the URL and must survive.
+    rf = tmp_path / "requirements.txt"
+    rf.write_text("git+https://host/r.git#subdirectory=pkg  # note\n")
+    assert parse_req_file(rf) == ["git+https://host/r.git#subdirectory=pkg"]
+
+
+def test_parse_req_file_preserves_double_dash_options(tmp_path):
```

Pylint (4.0.5)

[convention] Missing function or method docstring

(C0116)